source/backends/ubuntu/ubupkgindex.d

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ * Author: Iain Lane <iain.lane@canonical.com>
+ *
+ * Licensed under the GNU Lesser General Public License Version 3
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the license, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+module backends.ubuntu.ubupkgindex;
+
+import backends.debian.debpkg;
+import backends.debian.debpkgindex;
+import backends.interfaces;
+import backends.ubuntu.ubupkg;
+
+import std.container : Array, make;
+
+class UbuntuPackageIndex : DebianPackageIndex
+{
+
+public:
+    this (string dir)
+    {
+        /*
+         * UbuntuPackage needs to extract the langpacks, so we give it an array
+         * of all packages. We don't do this here, as you migh think makes
+         * sense, because it is a very expensive operation and we want to avoid
+         * doing it if it's not necessary (when no packages being processed are
+         * using langpacks).
+         */
+        allPackages = make!(Array!Package);
+        super (dir);
+    }
+
+    override DebPackage newPackage (string name, string ver, string arch)
+    {
+        return new UbuntuPackage (name, ver, arch, tmpDir, allPackages);
```

Comment:
Right on both points: We don't use `XDG_RUNTIME_DIR` unconditionally, because we might extract a lot of large objects, like images, which we don't clean up immediately (but all at once at the end of a generator run).
I'd rather like to explicitly use a tmpfs location when we know that the result will be small.

Pushing `tmpDir` in is fine, I think (if it isn't there, the package will use `$workspacedir/cache/tmp/$random/$name-$version`, where `$workspacedir/cache/tmp/$random` is always the same for each session. So in theory, I think, a package could also find out the locale location on its own).

Also: 921M!!! What is in there? That's an insane size for translations!